src/snowflake/snowpark/mock/_functions.py

```diff
@@ -523,26 +608,47 @@ def mock_to_timestamp(
 
         [ ] If the value is greater than or equal to 31536000000000000, then the value is treated as nanoseconds.
     """
+    # dateutil is a pandas dependency
+    import dateutil.parser
+
```

Comment:
If there's a better way to import this conditionally I'll fix this. Basically it should be available when pandas is available and as far as I know pandas is always available when local testing is running.

Reply:
I tried this out, but there's an issue with the latest version of dateutil being unavailable and I don't want to add a hard version dependency for it. For now I'll leave the local import so that pandas version is picked up.